testing_app/README.md

```diff
@@ -1,16 +1,36 @@
 # testing_app
 
-A new Flutter project.
+A Sample app that shows different types of testing in Flutter.
 
-## Getting Started
+This particular sample uses the [Provider][] package but any other state management approach
+would do.
 
-This project is a starting point for a Flutter application.
+[provider]: https://pub.dev/packages/provider
 
-A few resources to get you started if this is your first Flutter project:
+This sample is currently being built. Not all testing types are covered yet.
 
-- [Lab: Write your first Flutter app](https://flutter.dev/docs/get-started/codelab)
-- [Cookbook: Useful Flutter samples](https://flutter.dev/docs/cookbook)
+## Goals for this sample
 
-For help getting started with Flutter, view our
-[online documentation](https://flutter.dev/docs), which offers tutorials,
-samples, guidance on mobile development, and a full API reference.
+Show how to perform:
+
+- Widget Testing,
+- Flutter Driver(Integration) Testing,
+- Performance Testing, and
+- State Management Testing using the [Provider][] package.
+
+### How to run tests
+
+- Widget Tests:
+  - To run on physical device or emulator:
+    - Navigate to the project's root folder using command line.
+    - Run `flutter run test/<file_path>`
```

Comment:
Do you think it would be a good idea to have simple instructions on how to run different types of tests? Will add more in the future along with tests if we are keeping it.

Reply:
We will probably need a mini tutorial here explaining how this sample demonstrates different styles of testing, and how to run each one. Something to do once we have all the testing styles in place.

Reply:
Sure. I'll add that to the list of things to be done once all of the testing styles are in place.

Reply:
Maybe add a TODO here?